frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -289,6 +289,14 @@ fn construct_runtime_final_expansion(
 			type RuntimeBlock = #block;
 		}
 
+		trait InternalConstructRuntime {
+			#[inline(always)]
+			fn runtime_metadata(&self) -> #scrate::sp_std::vec::Vec<#scrate::metadata_ir::RuntimeApiMetadataIR> {
+				Default::default()
+			}
+		}
+		impl InternalConstructRuntime for &#name {}
+
```

Comment:
```suggestion
```

We don't need this? Do you try to circumvent the problem of not having `impl_runtime_apis!`?

Reply:
I should have read your pr description :P But this also tells us that this should be a comment in the code :D

Reply:
Have added a comment in the code, thanks!